// @mistralai/mistralai/models/components/agentscompletionrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentsCompletionRequest$ = exports.AgentsCompletionRequest$outboundSchema = exports.AgentsCompletionRequest$inboundSchema = exports.AgentsCompletionRequestToolChoice$ = exports.AgentsCompletionRequestToolChoice$outboundSchema = exports.AgentsCompletionRequestToolChoice$inboundSchema = exports.AgentsCompletionRequestMessages$ = exports.AgentsCompletionRequestMessages$outboundSchema = exports.AgentsCompletionRequestMessages$inboundSchema = exports.AgentsCompletionRequestStop$ = exports.AgentsCompletionRequestStop$outboundSchema = exports.AgentsCompletionRequestStop$inboundSchema = exports.AgentsCompletionRequestToolChoice = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const assistantmessage_js_1 = require("./assistantmessage.js");
const responseformat_js_1 = require("./responseformat.js");
const tool_js_1 = require("./tool.js");
const toolmessage_js_1 = require("./toolmessage.js");
const usermessage_js_1 = require("./usermessage.js");
const z = __importStar(require("zod"));
exports.AgentsCompletionRequestToolChoice = {
    Auto: "auto",
    None: "none",
    Any: "any",
};
/** @internal */
exports.AgentsCompletionRequestStop$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.AgentsCompletionRequestStop$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AgentsCompletionRequestStop$;
(function (AgentsCompletionRequestStop$) {
    /** @deprecated use `AgentsCompletionRequestStop$inboundSchema` instead. */
    AgentsCompletionRequestStop$.inboundSchema = exports.AgentsCompletionRequestStop$inboundSchema;
    /** @deprecated use `AgentsCompletionRequestStop$outboundSchema` instead. */
    AgentsCompletionRequestStop$.outboundSchema = exports.AgentsCompletionRequestStop$outboundSchema;
})(AgentsCompletionRequestStop$ || (exports.AgentsCompletionRequestStop$ = AgentsCompletionRequestStop$ = {}));
/** @internal */
exports.AgentsCompletionRequestMessages$inboundSchema = z.union([
    usermessage_js_1.UserMessage$inboundSchema.and(z.object({ role: z.literal("user") }).transform((v) => ({ role: v.role }))),
    assistantmessage_js_1.AssistantMessage$inboundSchema.and(z.object({ role: z.literal("assistant") }).transform((v) => ({ role: v.role }))),
    toolmessage_js_1.ToolMessage$inboundSchema.and(z.object({ role: z.literal("tool") }).transform((v) => ({ role: v.role }))),
]);
/** @internal */
exports.AgentsCompletionRequestMessages$outboundSchema = z.union([
    usermessage_js_1.UserMessage$outboundSchema.and(z.object({ role: z.literal("user") }).transform((v) => ({ role: v.role }))),
    assistantmessage_js_1.AssistantMessage$outboundSchema.and(z.object({ role: z.literal("assistant") }).transform((v) => ({ role: v.role }))),
    toolmessage_js_1.ToolMessage$outboundSchema.and(z.object({ role: z.literal("tool") }).transform((v) => ({ role: v.role }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AgentsCompletionRequestMessages$;
(function (AgentsCompletionRequestMessages$) {
    /** @deprecated use `AgentsCompletionRequestMessages$inboundSchema` instead. */
    AgentsCompletionRequestMessages$.inboundSchema = exports.AgentsCompletionRequestMessages$inboundSchema;
    /** @deprecated use `AgentsCompletionRequestMessages$outboundSchema` instead. */
    AgentsCompletionRequestMessages$.outboundSchema = exports.AgentsCompletionRequestMessages$outboundSchema;
})(AgentsCompletionRequestMessages$ || (exports.AgentsCompletionRequestMessages$ = AgentsCompletionRequestMessages$ = {}));
/** @internal */
exports.AgentsCompletionRequestToolChoice$inboundSchema = z.nativeEnum(exports.AgentsCompletionRequestToolChoice);
/** @internal */
exports.AgentsCompletionRequestToolChoice$outboundSchema = exports.AgentsCompletionRequestToolChoice$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AgentsCompletionRequestToolChoice$;
(function (AgentsCompletionRequestToolChoice$) {
    /** @deprecated use `AgentsCompletionRequestToolChoice$inboundSchema` instead. */
    AgentsCompletionRequestToolChoice$.inboundSchema = exports.AgentsCompletionRequestToolChoice$inboundSchema;
    /** @deprecated use `AgentsCompletionRequestToolChoice$outboundSchema` instead. */
    AgentsCompletionRequestToolChoice$.outboundSchema = exports.AgentsCompletionRequestToolChoice$outboundSchema;
})(AgentsCompletionRequestToolChoice$ || (exports.AgentsCompletionRequestToolChoice$ = AgentsCompletionRequestToolChoice$ = {}));
/** @internal */
exports.AgentsCompletionRequest$inboundSchema = z
    .object({
    max_tokens: z.nullable(z.number().int()).optional(),
    min_tokens: z.nullable(z.number().int()).optional(),
    stream: z.boolean().default(false),
    stop: z.union([z.string(), z.array(z.string())]).optional(),
    random_seed: z.nullable(z.number().int()).optional(),
    messages: z.array(z.union([
        usermessage_js_1.UserMessage$inboundSchema.and(z.object({ role: z.literal("user") }).transform((v) => ({ role: v.role }))),
        assistantmessage_js_1.AssistantMessage$inboundSchema.and(z.object({ role: z.literal("assistant") }).transform((v) => ({ role: v.role }))),
        toolmessage_js_1.ToolMessage$inboundSchema.and(z.object({ role: z.literal("tool") }).transform((v) => ({ role: v.role }))),
    ])),
    response_format: responseformat_js_1.ResponseFormat$inboundSchema.optional(),
    tools: z.nullable(z.array(tool_js_1.Tool$inboundSchema)).optional(),
    tool_choice: exports.AgentsCompletionRequestToolChoice$inboundSchema.default("auto"),
    agent_id: z.string(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        max_tokens: "maxTokens",
        min_tokens: "minTokens",
        random_seed: "randomSeed",
        response_format: "responseFormat",
        tool_choice: "toolChoice",
        agent_id: "agentId",
    });
});
/** @internal */
exports.AgentsCompletionRequest$outboundSchema = z
    .object({
    maxTokens: z.nullable(z.number().int()).optional(),
    minTokens: z.nullable(z.number().int()).optional(),
    stream: z.boolean().default(false),
    stop: z.union([z.string(), z.array(z.string())]).optional(),
    randomSeed: z.nullable(z.number().int()).optional(),
    messages: z.array(z.union([
        usermessage_js_1.UserMessage$outboundSchema.and(z.object({ role: z.literal("user") }).transform((v) => ({ role: v.role }))),
        assistantmessage_js_1.AssistantMessage$outboundSchema.and(z.object({ role: z.literal("assistant") }).transform((v) => ({ role: v.role }))),
        toolmessage_js_1.ToolMessage$outboundSchema.and(z.object({ role: z.literal("tool") }).transform((v) => ({ role: v.role }))),
    ])),
    responseFormat: responseformat_js_1.ResponseFormat$outboundSchema.optional(),
    tools: z.nullable(z.array(tool_js_1.Tool$outboundSchema)).optional(),
    toolChoice: exports.AgentsCompletionRequestToolChoice$outboundSchema.default("auto"),
    agentId: z.string(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        maxTokens: "max_tokens",
        minTokens: "min_tokens",
        randomSeed: "random_seed",
        responseFormat: "response_format",
        toolChoice: "tool_choice",
        agentId: "agent_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AgentsCompletionRequest$;
(function (AgentsCompletionRequest$) {
    /** @deprecated use `AgentsCompletionRequest$inboundSchema` instead. */
    AgentsCompletionRequest$.inboundSchema = exports.AgentsCompletionRequest$inboundSchema;
    /** @deprecated use `AgentsCompletionRequest$outboundSchema` instead. */
    AgentsCompletionRequest$.outboundSchema = exports.AgentsCompletionRequest$outboundSchema;
})(AgentsCompletionRequest$ || (exports.AgentsCompletionRequest$ = AgentsCompletionRequest$ = {}));
