// @mistralai/mistralai/models/components/chatcompletionchoice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatCompletionChoice$ = exports.ChatCompletionChoice$outboundSchema = exports.ChatCompletionChoice$inboundSchema = exports.FinishReason$ = exports.FinishReason$outboundSchema = exports.FinishReason$inboundSchema = exports.FinishReason = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const assistantmessage_js_1 = require("./assistantmessage.js");
const z = __importStar(require("zod"));
exports.FinishReason = {
    Stop: "stop",
    Length: "length",
    ModelLength: "model_length",
    Error: "error",
    ToolCalls: "tool_calls",
};
/** @internal */
exports.FinishReason$inboundSchema = z.nativeEnum(exports.FinishReason);
/** @internal */
exports.FinishReason$outboundSchema = exports.FinishReason$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FinishReason$;
(function (FinishReason$) {
    /** @deprecated use `FinishReason$inboundSchema` instead. */
    FinishReason$.inboundSchema = exports.FinishReason$inboundSchema;
    /** @deprecated use `FinishReason$outboundSchema` instead. */
    FinishReason$.outboundSchema = exports.FinishReason$outboundSchema;
})(FinishReason$ || (exports.FinishReason$ = FinishReason$ = {}));
/** @internal */
exports.ChatCompletionChoice$inboundSchema = z
    .object({
    index: z.number().int(),
    message: assistantmessage_js_1.AssistantMessage$inboundSchema.optional(),
    finish_reason: exports.FinishReason$inboundSchema,
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        finish_reason: "finishReason",
    });
});
/** @internal */
exports.ChatCompletionChoice$outboundSchema = z
    .object({
    index: z.number().int(),
    message: assistantmessage_js_1.AssistantMessage$outboundSchema.optional(),
    finishReason: exports.FinishReason$outboundSchema,
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        finishReason: "finish_reason",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ChatCompletionChoice$;
(function (ChatCompletionChoice$) {
    /** @deprecated use `ChatCompletionChoice$inboundSchema` instead. */
    ChatCompletionChoice$.inboundSchema = exports.ChatCompletionChoice$inboundSchema;
    /** @deprecated use `ChatCompletionChoice$outboundSchema` instead. */
    ChatCompletionChoice$.outboundSchema = exports.ChatCompletionChoice$outboundSchema;
})(ChatCompletionChoice$ || (exports.ChatCompletionChoice$ = ChatCompletionChoice$ = {}));
