// @mistralai/mistralai/models/components/deltamessage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeltaMessage$ = exports.DeltaMessage$outboundSchema = exports.DeltaMessage$inboundSchema = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const toolcall_js_1 = require("./toolcall.js");
const z = __importStar(require("zod"));
/** @internal */
exports.DeltaMessage$inboundSchema = z
    .object({
    role: z.string().optional(),
    content: z.string().optional(),
    tool_calls: z.nullable(z.array(toolcall_js_1.ToolCall$inboundSchema)).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        tool_calls: "toolCalls",
    });
});
/** @internal */
exports.DeltaMessage$outboundSchema = z
    .object({
    role: z.string().optional(),
    content: z.string().optional(),
    toolCalls: z.nullable(z.array(toolcall_js_1.ToolCall$outboundSchema)).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        toolCalls: "tool_calls",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeltaMessage$;
(function (DeltaMessage$) {
    /** @deprecated use `DeltaMessage$inboundSchema` instead. */
    DeltaMessage$.inboundSchema = exports.DeltaMessage$inboundSchema;
    /** @deprecated use `DeltaMessage$outboundSchema` instead. */
    DeltaMessage$.outboundSchema = exports.DeltaMessage$outboundSchema;
})(DeltaMessage$ || (exports.DeltaMessage$ = DeltaMessage$ = {}));
