// @mistralai/mistralai/models/components/detailedjobout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetailedJobOut$ = exports.DetailedJobOut$outboundSchema = exports.DetailedJobOut$inboundSchema = exports.DetailedJobOutStatus$ = exports.DetailedJobOutStatus$outboundSchema = exports.DetailedJobOutStatus$inboundSchema = exports.DetailedJobOutStatus = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const checkpointout_js_1 = require("./checkpointout.js");
const eventout_js_1 = require("./eventout.js");
const finetuneablemodel_js_1 = require("./finetuneablemodel.js");
const githubrepositoryout_js_1 = require("./githubrepositoryout.js");
const jobmetadataout_js_1 = require("./jobmetadataout.js");
const trainingparameters_js_1 = require("./trainingparameters.js");
const wandbintegrationout_js_1 = require("./wandbintegrationout.js");
const z = __importStar(require("zod"));
exports.DetailedJobOutStatus = {
    Queued: "QUEUED",
    Started: "STARTED",
    Validating: "VALIDATING",
    Validated: "VALIDATED",
    Running: "RUNNING",
    FailedValidation: "FAILED_VALIDATION",
    Failed: "FAILED",
    Success: "SUCCESS",
    Cancelled: "CANCELLED",
    CancellationRequested: "CANCELLATION_REQUESTED",
};
/** @internal */
exports.DetailedJobOutStatus$inboundSchema = z.nativeEnum(exports.DetailedJobOutStatus);
/** @internal */
exports.DetailedJobOutStatus$outboundSchema = exports.DetailedJobOutStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOutStatus$;
(function (DetailedJobOutStatus$) {
    /** @deprecated use `DetailedJobOutStatus$inboundSchema` instead. */
    DetailedJobOutStatus$.inboundSchema = exports.DetailedJobOutStatus$inboundSchema;
    /** @deprecated use `DetailedJobOutStatus$outboundSchema` instead. */
    DetailedJobOutStatus$.outboundSchema = exports.DetailedJobOutStatus$outboundSchema;
})(DetailedJobOutStatus$ || (exports.DetailedJobOutStatus$ = DetailedJobOutStatus$ = {}));
/** @internal */
exports.DetailedJobOut$inboundSchema = z
    .object({
    id: z.string(),
    auto_start: z.boolean(),
    hyperparameters: trainingparameters_js_1.TrainingParameters$inboundSchema,
    model: finetuneablemodel_js_1.FineTuneableModel$inboundSchema,
    status: exports.DetailedJobOutStatus$inboundSchema,
    job_type: z.string(),
    created_at: z.number().int(),
    modified_at: z.number().int(),
    training_files: z.array(z.string()),
    validation_files: z.nullable(z.array(z.string())).optional(),
    object: z.literal("job"),
    fine_tuned_model: z.nullable(z.string()).optional(),
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegrationout_js_1.WandbIntegrationOut$inboundSchema)).optional(),
    trained_tokens: z.nullable(z.number().int()).optional(),
    repositories: z.array(githubrepositoryout_js_1.GithubRepositoryOut$inboundSchema).optional(),
    metadata: z.nullable(jobmetadataout_js_1.JobMetadataOut$inboundSchema).optional(),
    events: z.array(eventout_js_1.EventOut$inboundSchema).optional(),
    checkpoints: z.array(checkpointout_js_1.CheckpointOut$inboundSchema).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        auto_start: "autoStart",
        job_type: "jobType",
        created_at: "createdAt",
        modified_at: "modifiedAt",
        training_files: "trainingFiles",
        validation_files: "validationFiles",
        fine_tuned_model: "fineTunedModel",
        trained_tokens: "trainedTokens",
    });
});
/** @internal */
exports.DetailedJobOut$outboundSchema = z
    .object({
    id: z.string(),
    autoStart: z.boolean(),
    hyperparameters: trainingparameters_js_1.TrainingParameters$outboundSchema,
    model: finetuneablemodel_js_1.FineTuneableModel$outboundSchema,
    status: exports.DetailedJobOutStatus$outboundSchema,
    jobType: z.string(),
    createdAt: z.number().int(),
    modifiedAt: z.number().int(),
    trainingFiles: z.array(z.string()),
    validationFiles: z.nullable(z.array(z.string())).optional(),
    object: z.literal("job").default("job"),
    fineTunedModel: z.nullable(z.string()).optional(),
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegrationout_js_1.WandbIntegrationOut$outboundSchema)).optional(),
    trainedTokens: z.nullable(z.number().int()).optional(),
    repositories: z.array(githubrepositoryout_js_1.GithubRepositoryOut$outboundSchema).optional(),
    metadata: z.nullable(jobmetadataout_js_1.JobMetadataOut$outboundSchema).optional(),
    events: z.array(eventout_js_1.EventOut$outboundSchema).optional(),
    checkpoints: z.array(checkpointout_js_1.CheckpointOut$outboundSchema).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        autoStart: "auto_start",
        jobType: "job_type",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trainingFiles: "training_files",
        validationFiles: "validation_files",
        fineTunedModel: "fine_tuned_model",
        trainedTokens: "trained_tokens",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOut$;
(function (DetailedJobOut$) {
    /** @deprecated use `DetailedJobOut$inboundSchema` instead. */
    DetailedJobOut$.inboundSchema = exports.DetailedJobOut$inboundSchema;
    /** @deprecated use `DetailedJobOut$outboundSchema` instead. */
    DetailedJobOut$.outboundSchema = exports.DetailedJobOut$outboundSchema;
})(DetailedJobOut$ || (exports.DetailedJobOut$ = DetailedJobOut$ = {}));
