// @mistralai/mistralai/models/components/embeddingrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmbeddingRequest$ = exports.EmbeddingRequest$outboundSchema = exports.EmbeddingRequest$inboundSchema = exports.Inputs$ = exports.Inputs$outboundSchema = exports.Inputs$inboundSchema = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
exports.Inputs$inboundSchema = z.union([
    z.string(),
    z.array(z.string()),
]);
/** @internal */
exports.Inputs$outboundSchema = z.union([
    z.string(),
    z.array(z.string()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Inputs$;
(function (Inputs$) {
    /** @deprecated use `Inputs$inboundSchema` instead. */
    Inputs$.inboundSchema = exports.Inputs$inboundSchema;
    /** @deprecated use `Inputs$outboundSchema` instead. */
    Inputs$.outboundSchema = exports.Inputs$outboundSchema;
})(Inputs$ || (exports.Inputs$ = Inputs$ = {}));
/** @internal */
exports.EmbeddingRequest$inboundSchema = z
    .object({
    input: z.union([z.string(), z.array(z.string())]),
    model: z.string(),
    encoding_format: z.nullable(z.string()).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        input: "inputs",
        encoding_format: "encodingFormat",
    });
});
/** @internal */
exports.EmbeddingRequest$outboundSchema = z
    .object({
    inputs: z.union([z.string(), z.array(z.string())]),
    model: z.string(),
    encodingFormat: z.nullable(z.string()).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        inputs: "input",
        encodingFormat: "encoding_format",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EmbeddingRequest$;
(function (EmbeddingRequest$) {
    /** @deprecated use `EmbeddingRequest$inboundSchema` instead. */
    EmbeddingRequest$.inboundSchema = exports.EmbeddingRequest$inboundSchema;
    /** @deprecated use `EmbeddingRequest$outboundSchema` instead. */
    EmbeddingRequest$.outboundSchema = exports.EmbeddingRequest$outboundSchema;
})(EmbeddingRequest$ || (exports.EmbeddingRequest$ = EmbeddingRequest$ = {}));
