// @mistralai/mistralai/models/components/jobin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobIn$ = exports.JobIn$outboundSchema = exports.JobIn$inboundSchema = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const finetuneablemodel_js_1 = require("./finetuneablemodel.js");
const githubrepositoryin_js_1 = require("./githubrepositoryin.js");
const trainingfile_js_1 = require("./trainingfile.js");
const trainingparametersin_js_1 = require("./trainingparametersin.js");
const wandbintegration_js_1 = require("./wandbintegration.js");
const z = __importStar(require("zod"));
/** @internal */
exports.JobIn$inboundSchema = z
    .object({
    model: finetuneablemodel_js_1.FineTuneableModel$inboundSchema,
    training_files: z.array(trainingfile_js_1.TrainingFile$inboundSchema).optional(),
    validation_files: z.nullable(z.array(z.string())).optional(),
    hyperparameters: trainingparametersin_js_1.TrainingParametersIn$inboundSchema,
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegration_js_1.WandbIntegration$inboundSchema)).optional(),
    repositories: z.array(githubrepositoryin_js_1.GithubRepositoryIn$inboundSchema).optional(),
    auto_start: z.boolean().optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        training_files: "trainingFiles",
        validation_files: "validationFiles",
        auto_start: "autoStart",
    });
});
/** @internal */
exports.JobIn$outboundSchema = z
    .object({
    model: finetuneablemodel_js_1.FineTuneableModel$outboundSchema,
    trainingFiles: z.array(trainingfile_js_1.TrainingFile$outboundSchema).optional(),
    validationFiles: z.nullable(z.array(z.string())).optional(),
    hyperparameters: trainingparametersin_js_1.TrainingParametersIn$outboundSchema,
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegration_js_1.WandbIntegration$outboundSchema)).optional(),
    repositories: z.array(githubrepositoryin_js_1.GithubRepositoryIn$outboundSchema).optional(),
    autoStart: z.boolean().optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        trainingFiles: "training_files",
        validationFiles: "validation_files",
        autoStart: "auto_start",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JobIn$;
(function (JobIn$) {
    /** @deprecated use `JobIn$inboundSchema` instead. */
    JobIn$.inboundSchema = exports.JobIn$inboundSchema;
    /** @deprecated use `JobIn$outboundSchema` instead. */
    JobIn$.outboundSchema = exports.JobIn$outboundSchema;
})(JobIn$ || (exports.JobIn$ = JobIn$ = {}));
