// @mistralai/mistralai/models/components/jobout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobOut$ = exports.JobOut$outboundSchema = exports.JobOut$inboundSchema = exports.Status$ = exports.Status$outboundSchema = exports.Status$inboundSchema = exports.Status = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const finetuneablemodel_js_1 = require("./finetuneablemodel.js");
const githubrepositoryout_js_1 = require("./githubrepositoryout.js");
const jobmetadataout_js_1 = require("./jobmetadataout.js");
const trainingparameters_js_1 = require("./trainingparameters.js");
const wandbintegrationout_js_1 = require("./wandbintegrationout.js");
const z = __importStar(require("zod"));
/**
 * The current status of the fine-tuning job.
 */
exports.Status = {
    Queued: "QUEUED",
    Started: "STARTED",
    Validating: "VALIDATING",
    Validated: "VALIDATED",
    Running: "RUNNING",
    FailedValidation: "FAILED_VALIDATION",
    Failed: "FAILED",
    Success: "SUCCESS",
    Cancelled: "CANCELLED",
    CancellationRequested: "CANCELLATION_REQUESTED",
};
/** @internal */
exports.Status$inboundSchema = z.nativeEnum(exports.Status);
/** @internal */
exports.Status$outboundSchema = exports.Status$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Status$;
(function (Status$) {
    /** @deprecated use `Status$inboundSchema` instead. */
    Status$.inboundSchema = exports.Status$inboundSchema;
    /** @deprecated use `Status$outboundSchema` instead. */
    Status$.outboundSchema = exports.Status$outboundSchema;
})(Status$ || (exports.Status$ = Status$ = {}));
/** @internal */
exports.JobOut$inboundSchema = z
    .object({
    id: z.string(),
    auto_start: z.boolean(),
    hyperparameters: trainingparameters_js_1.TrainingParameters$inboundSchema,
    model: finetuneablemodel_js_1.FineTuneableModel$inboundSchema,
    status: exports.Status$inboundSchema,
    job_type: z.string(),
    created_at: z.number().int(),
    modified_at: z.number().int(),
    training_files: z.array(z.string()),
    validation_files: z.nullable(z.array(z.string())).optional(),
    object: z.literal("job"),
    fine_tuned_model: z.nullable(z.string()).optional(),
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegrationout_js_1.WandbIntegrationOut$inboundSchema)).optional(),
    trained_tokens: z.nullable(z.number().int()).optional(),
    repositories: z.array(githubrepositoryout_js_1.GithubRepositoryOut$inboundSchema).optional(),
    metadata: z.nullable(jobmetadataout_js_1.JobMetadataOut$inboundSchema).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        auto_start: "autoStart",
        job_type: "jobType",
        created_at: "createdAt",
        modified_at: "modifiedAt",
        training_files: "trainingFiles",
        validation_files: "validationFiles",
        fine_tuned_model: "fineTunedModel",
        trained_tokens: "trainedTokens",
    });
});
/** @internal */
exports.JobOut$outboundSchema = z
    .object({
    id: z.string(),
    autoStart: z.boolean(),
    hyperparameters: trainingparameters_js_1.TrainingParameters$outboundSchema,
    model: finetuneablemodel_js_1.FineTuneableModel$outboundSchema,
    status: exports.Status$outboundSchema,
    jobType: z.string(),
    createdAt: z.number().int(),
    modifiedAt: z.number().int(),
    trainingFiles: z.array(z.string()),
    validationFiles: z.nullable(z.array(z.string())).optional(),
    object: z.literal("job").default("job"),
    fineTunedModel: z.nullable(z.string()).optional(),
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegrationout_js_1.WandbIntegrationOut$outboundSchema)).optional(),
    trainedTokens: z.nullable(z.number().int()).optional(),
    repositories: z.array(githubrepositoryout_js_1.GithubRepositoryOut$outboundSchema).optional(),
    metadata: z.nullable(jobmetadataout_js_1.JobMetadataOut$outboundSchema).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        autoStart: "auto_start",
        jobType: "job_type",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trainingFiles: "training_files",
        validationFiles: "validation_files",
        fineTunedModel: "fine_tuned_model",
        trainedTokens: "trained_tokens",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JobOut$;
(function (JobOut$) {
    /** @deprecated use `JobOut$inboundSchema` instead. */
    JobOut$.inboundSchema = exports.JobOut$inboundSchema;
    /** @deprecated use `JobOut$outboundSchema` instead. */
    JobOut$.outboundSchema = exports.JobOut$outboundSchema;
})(JobOut$ || (exports.JobOut$ = JobOut$ = {}));
