// @mistralai/mistralai/models/components/modelcard.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelCard$ = exports.ModelCard$outboundSchema = exports.ModelCard$inboundSchema = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const modelcapabilities_js_1 = require("./modelcapabilities.js");
const z = __importStar(require("zod"));
/** @internal */
exports.ModelCard$inboundSchema = z
    .object({
    id: z.string(),
    object: z.string().default("model"),
    created: z.number().int().optional(),
    owned_by: z.string().default("mistralai"),
    root: z.nullable(z.string()).optional(),
    archived: z.boolean().default(false),
    name: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    capabilities: modelcapabilities_js_1.ModelCapabilities$inboundSchema,
    max_context_length: z.number().int().default(32768),
    aliases: z.array(z.string()).optional(),
    deprecation: z
        .nullable(z
        .string()
        .datetime({ offset: true })
        .transform((v) => new Date(v)))
        .optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        owned_by: "ownedBy",
        max_context_length: "maxContextLength",
    });
});
/** @internal */
exports.ModelCard$outboundSchema = z
    .object({
    id: z.string(),
    object: z.string().default("model"),
    created: z.number().int().optional(),
    ownedBy: z.string().default("mistralai"),
    root: z.nullable(z.string()).optional(),
    archived: z.boolean().default(false),
    name: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    capabilities: modelcapabilities_js_1.ModelCapabilities$outboundSchema,
    maxContextLength: z.number().int().default(32768),
    aliases: z.array(z.string()).optional(),
    deprecation: z.nullable(z.date().transform((v) => v.toISOString())).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        ownedBy: "owned_by",
        maxContextLength: "max_context_length",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ModelCard$;
(function (ModelCard$) {
    /** @deprecated use `ModelCard$inboundSchema` instead. */
    ModelCard$.inboundSchema = exports.ModelCard$inboundSchema;
    /** @deprecated use `ModelCard$outboundSchema` instead. */
    ModelCard$.outboundSchema = exports.ModelCard$outboundSchema;
})(ModelCard$ || (exports.ModelCard$ = ModelCard$ = {}));
