// @mistralai/mistralai/models/components/systemmessage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemMessage$ = exports.SystemMessage$outboundSchema = exports.SystemMessage$inboundSchema = exports.Role$ = exports.Role$outboundSchema = exports.Role$inboundSchema = exports.Content$ = exports.Content$outboundSchema = exports.Content$inboundSchema = exports.Role = void 0;
const contentchunk_js_1 = require("./contentchunk.js");
const z = __importStar(require("zod"));
exports.Role = {
    System: "system",
};
/** @internal */
exports.Content$inboundSchema = z.union([
    z.string(),
    z.array(contentchunk_js_1.ContentChunk$inboundSchema),
]);
/** @internal */
exports.Content$outboundSchema = z.union([
    z.string(),
    z.array(contentchunk_js_1.ContentChunk$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Content$;
(function (Content$) {
    /** @deprecated use `Content$inboundSchema` instead. */
    Content$.inboundSchema = exports.Content$inboundSchema;
    /** @deprecated use `Content$outboundSchema` instead. */
    Content$.outboundSchema = exports.Content$outboundSchema;
})(Content$ || (exports.Content$ = Content$ = {}));
/** @internal */
exports.Role$inboundSchema = z.nativeEnum(exports.Role);
/** @internal */
exports.Role$outboundSchema = exports.Role$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Role$;
(function (Role$) {
    /** @deprecated use `Role$inboundSchema` instead. */
    Role$.inboundSchema = exports.Role$inboundSchema;
    /** @deprecated use `Role$outboundSchema` instead. */
    Role$.outboundSchema = exports.Role$outboundSchema;
})(Role$ || (exports.Role$ = Role$ = {}));
/** @internal */
exports.SystemMessage$inboundSchema = z.object({
    content: z.union([z.string(), z.array(contentchunk_js_1.ContentChunk$inboundSchema)]),
    role: exports.Role$inboundSchema.default("system"),
});
/** @internal */
exports.SystemMessage$outboundSchema = z.object({
    content: z.union([z.string(), z.array(contentchunk_js_1.ContentChunk$outboundSchema)]),
    role: exports.Role$outboundSchema.default("system"),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SystemMessage$;
(function (SystemMessage$) {
    /** @deprecated use `SystemMessage$inboundSchema` instead. */
    SystemMessage$.inboundSchema = exports.SystemMessage$inboundSchema;
    /** @deprecated use `SystemMessage$outboundSchema` instead. */
    SystemMessage$.outboundSchema = exports.SystemMessage$outboundSchema;
})(SystemMessage$ || (exports.SystemMessage$ = SystemMessage$ = {}));
