// @mistralai/mistralai/models/components/trainingparametersin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrainingParametersIn$ = exports.TrainingParametersIn$outboundSchema = exports.TrainingParametersIn$inboundSchema = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
exports.TrainingParametersIn$inboundSchema = z
    .object({
    training_steps: z.nullable(z.number().int()).optional(),
    learning_rate: z.number().default(0.0001),
    epochs: z.nullable(z.number()).optional(),
    fim_ratio: z.nullable(z.number()).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        training_steps: "trainingSteps",
        learning_rate: "learningRate",
        fim_ratio: "fimRatio",
    });
});
/** @internal */
exports.TrainingParametersIn$outboundSchema = z
    .object({
    trainingSteps: z.nullable(z.number().int()).optional(),
    learningRate: z.number().default(0.0001),
    epochs: z.nullable(z.number()).optional(),
    fimRatio: z.nullable(z.number()).optional(),
})
    .transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        trainingSteps: "training_steps",
        learningRate: "learning_rate",
        fimRatio: "fim_ratio",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrainingParametersIn$;
(function (TrainingParametersIn$) {
    /** @deprecated use `TrainingParametersIn$inboundSchema` instead. */
    TrainingParametersIn$.inboundSchema = exports.TrainingParametersIn$inboundSchema;
    /** @deprecated use `TrainingParametersIn$outboundSchema` instead. */
    TrainingParametersIn$.outboundSchema = exports.TrainingParametersIn$outboundSchema;
})(TrainingParametersIn$ || (exports.TrainingParametersIn$ = TrainingParametersIn$ = {}));
