// @mistralai/mistralai/models/components/unarchiveftmodelout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnarchiveFTModelOut$ = exports.UnarchiveFTModelOut$outboundSchema = exports.UnarchiveFTModelOut$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.UnarchiveFTModelOut$inboundSchema = z.object({
    id: z.string(),
    object: z.literal("model"),
    archived: z.boolean().default(false),
});
/** @internal */
exports.UnarchiveFTModelOut$outboundSchema = z.object({
    id: z.string(),
    object: z.literal("model").default("model"),
    archived: z.boolean().default(false),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UnarchiveFTModelOut$;
(function (UnarchiveFTModelOut$) {
    /** @deprecated use `UnarchiveFTModelOut$inboundSchema` instead. */
    UnarchiveFTModelOut$.inboundSchema = exports.UnarchiveFTModelOut$inboundSchema;
    /** @deprecated use `UnarchiveFTModelOut$outboundSchema` instead. */
    UnarchiveFTModelOut$.outboundSchema = exports.UnarchiveFTModelOut$outboundSchema;
})(UnarchiveFTModelOut$ || (exports.UnarchiveFTModelOut$ = UnarchiveFTModelOut$ = {}));
