// @mistralai/mistralai/models/operations/filesapiroutesuploadfile.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesApiRoutesUploadFileMultiPartBodyParams$ = exports.FilesApiRoutesUploadFileMultiPartBodyParams$outboundSchema = exports.FilesApiRoutesUploadFileMultiPartBodyParams$inboundSchema = exports.FileT$ = exports.FileT$outboundSchema = exports.FileT$inboundSchema = void 0;
const b64$ = __importStar(require("../../lib/base64.js"));
const blobs_js_1 = require("../../types/blobs.js");
const z = __importStar(require("zod"));
/** @internal */
exports.FileT$inboundSchema = z.object({
    fileName: z.string(),
    content: b64$.zodInbound,
});
/** @internal */
exports.FileT$outboundSchema = z.object({
    fileName: z.string(),
    content: b64$.zodOutbound,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FileT$;
(function (FileT$) {
    /** @deprecated use `FileT$inboundSchema` instead. */
    FileT$.inboundSchema = exports.FileT$inboundSchema;
    /** @deprecated use `FileT$outboundSchema` instead. */
    FileT$.outboundSchema = exports.FileT$outboundSchema;
})(FileT$ || (exports.FileT$ = FileT$ = {}));
/** @internal */
exports.FilesApiRoutesUploadFileMultiPartBodyParams$inboundSchema = z.object({
    file: z.lazy(() => exports.FileT$inboundSchema),
    purpose: z.literal("fine-tune"),
});
/** @internal */
exports.FilesApiRoutesUploadFileMultiPartBodyParams$outboundSchema = z.object({
    file: z.lazy(() => exports.FileT$outboundSchema).or(blobs_js_1.blobLikeSchema),
    purpose: z.literal("fine-tune").default("fine-tune"),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesApiRoutesUploadFileMultiPartBodyParams$;
(function (FilesApiRoutesUploadFileMultiPartBodyParams$) {
    /** @deprecated use `FilesApiRoutesUploadFileMultiPartBodyParams$inboundSchema` instead. */
    FilesApiRoutesUploadFileMultiPartBodyParams$.inboundSchema = exports.FilesApiRoutesUploadFileMultiPartBodyParams$inboundSchema;
    /** @deprecated use `FilesApiRoutesUploadFileMultiPartBodyParams$outboundSchema` instead. */
    FilesApiRoutesUploadFileMultiPartBodyParams$.outboundSchema = exports.FilesApiRoutesUploadFileMultiPartBodyParams$outboundSchema;
})(FilesApiRoutesUploadFileMultiPartBodyParams$ || (exports.FilesApiRoutesUploadFileMultiPartBodyParams$ = FilesApiRoutesUploadFileMultiPartBodyParams$ = {}));
