// @mistralai/mistralai/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.ServerList = exports.ServerProd = void 0;
exports.serverURLFromOptions = serverURLFromOptions;
const url_js_1 = require("./url.js");
/**
 * Production server
 */
exports.ServerProd = "prod";
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = {
    [exports.ServerProd]: "https://api.mistral.ai",
};
function serverURLFromOptions(options) {
    var _a;
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = (_a = options.server) !== null && _a !== void 0 ? _a : exports.ServerProd;
        serverURL = exports.ServerList[server] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "0.0.2",
    sdkVersion: "1.0.2",
    genVersion: "2.390.6",
    userAgent: "speakeasy-sdk/typescript 1.0.2 2.390.6 0.0.2 @mistralai/mistralai",
};
